frappe/core/doctype/submission_queue/submission_queue.py

```diff
@@ -0,0 +1,197 @@
+		from frappe.query_builder import Interval
+		from frappe.query_builder.functions import Now
+
+		table = frappe.qb.DocType("Submission Queue")
+		frappe.db.delete(table, filters=(table.modified < (Now() - Interval(days=days))))
+
+	def insert(self, to_be_queued_doc: Document, action: str):
+		self.to_be_queued_doc = to_be_queued_doc
+		self.action_for_queuing = action
+		super().insert(ignore_permissions=True)
+
+	def lock(self):
+		self.queued_doc.lock()
+
+	def unlock(self):
+		self.queued_doc.unlock()
+
+	def after_insert(self):
+		self.queue_action(
+			"background_submission",
+			to_be_queued_doc=self.queued_doc,
+			action_for_queuing=self.action_for_queuing,
+			timeout=600,
+			enqueue_after_commit=True,
+		)
+
+	def background_submission(self, to_be_queued_doc: Document, action_for_queuing: str):
+		current_job = get_current_job()
+
+		# Set the job id for that submission doctype
+		frappe.db.set_value(
+			self.doctype,
+			self.name,
+			{"job_id": current_job.id},
+			update_modified=False,
+		)
+		_action = action_for_queuing.lower()
+		if _action == "update":
+			_action = "submit"
+
+		try:
+			getattr(to_be_queued_doc, _action)()
+			add_data_to_monitor(
+				doctype=to_be_queued_doc.doctype,
+				docname=to_be_queued_doc.name,
+				action=_action,
+				execution_time=time_diff_in_seconds(now(), self.created_at),
+				enqueued_by=self.enqueued_by,
+			)
+			values = {"status": "Finished"}
+		except Exception:
+			values = {"status": "Failed", "exception": frappe.get_traceback()}
+			frappe.db.rollback()
+
+		values["ended_at"] = now()
+		frappe.db.set_value(self.doctype, self.name, values, update_modified=False)
+		self.notify(values["status"], action_for_queuing)
+
+	def notify(self, submission_status: str, action: str):
+		if submission_status == "Failed":
+			doctype = self.doctype
+			docname = self.name
+			message = _("Submission of {0} {1} with action {2} failed")
+		else:
+			doctype = self.ref_doctype
+			docname = self.ref_docname
+			message = _("Submission of {0} {1} with action {2} completed successfully")
+
+		message = message.format(
+			frappe.bold(str(self.ref_doctype)), frappe.bold(self.ref_docname), frappe.bold(action)
+		)
+		time_diff = time_diff_in_seconds(now(), self.created_at)
+		if cint(time_diff) <= 60:
+			frappe.publish_realtime(
+				"msgprint",
+				{
+					"message": message
+					+ f". View it <a href='/app/{doctype.lower().replace(' ', '-')}/{docname}'><b>here</b></a>"
+					+ f". Execution time {round(float(time_diff), 2)} seconds",
+					"alert": True,
+					"indicator": "red" if submission_status == "Failed" else "green",
+				},
+				user=self.enqueued_by,
+			)
+		else:
+			notification_doc = {
+				"type": "Alert",
+				"document_type": doctype,
+				"document_name": docname,
+				"subject": message,
+			}
+
+			notify_to = frappe.db.get_value("User", self.enqueued_by, fieldname="email")
+			enqueue_create_notification([notify_to], notification_doc)
+
+	def _unlock_reference_doc(self):
```

Comment:
this whole method is not working - and gives an "UnboundLocalError" error - since there is no status variable defined in the case there is `job_id` found :(

Reply:
Fixed